tests/Game.class.test.js

```diff
@@ -0,0 +1,965 @@
+
+      game2048.start();
+      game2048.moveLeft();
+
+      const state = game2048.getState();
+
+      expect(state[0][0]).not.toBe(0);
+      expect(state[1][0]).not.toBe(0);
+      expect(state[2][0]).not.toBe(0);
+      expect(state[3][0]).not.toBe(0);
+    });
+
+    it('should add a new cell with 2 or 4', () => {
+      const game2048 = new Game([
+        [8, 0, 16, 0],
+        [0, 16, 0, 32],
+        [8, 0, 32, 0],
+        [0, 16, 0, 64],
+      ]);
+
+      game2048.start();
+
+      const sumBeforeTheMove = sum(game2048);
+
+      game2048.moveLeft();
+
+      expect(count(game2048)).toBe(11);
+      expect(sum(game2048)).toBeGreaterThanOrEqual(sumBeforeTheMove + 2);
+      expect(sum(game2048)).toBeLessThanOrEqual(sumBeforeTheMove + 4);
+    });
+
+    it('should stack cells to the left without merging', () => {
+      const game2048 = new Game([
+        [0, 16, 0, 8],
+        [8, 0, 16, 0],
+        [0, 8, 0, 32],
+        [32, 0, 8, 0],
+      ]);
+
+      game2048.start();
+
+      const rowsBefore = game2048.getState();
+
+      game2048.moveLeft();
+
+      const rows = game2048.getState();
+
+      for (let i = 0; i < 4; i++) {
+        const nonEmptyCells = rowsBefore[i].filter(Boolean);
+
+        expect(rows[i].slice(0, nonEmptyCells.length))
+          .toEqual(nonEmptyCells);
+      }
+    });
+
+    it('should merge cells with the same value', () => {
+      const game2048 = new Game([
+        [16, 8, 0, 0],
+        [0, 0, 8, 8],
+        [0, 8, 8, 0],
+        [8, 0, 0, 16],
+      ]);
+
+      game2048.start();
+      game2048.moveLeft();
+
+      game2048.getState().forEach(row => {
+        expect(row).toContain(16);
+      });
+    });
+
+    it('should merge multiple cells with the same value to the left', () => {
+      const game2048 = new Game([
+        [8, 8, 16, 16],
+        [16, 16, 8, 8],
+        [32, 32, 0, 0],
+        [16, 16, 64, 64],
+      ]);
+
+      game2048.start();
+      game2048.moveLeft();
+
+      const state = game2048.getState();
+
+      expect(state[0][0]).toBe(16);
+      expect(state[0][1]).toBe(32);
+      expect(state[1][0]).toBe(32);
+      expect(state[1][1]).toBe(16);
+      expect(state[2][0]).toBe(64);
+      expect(state[3][0]).toBe(32);
+      expect(state[3][1]).toBe(128);
+    });
+
+    it('should not merge multiple cells with the same value to the left multiple times during the move', () => {
+      const game2048 = new Game([
+        [8, 8, 0, 0],
+        [16, 16, 16, 16],
+        [32, 32, 64, 128],
+        [2, 2, 4, 8],
+      ]);
+
+      game2048.start();
+      game2048.moveLeft();
+
+      const state = game2048.getState();
+
+      expect(state[0][0]).toBe(16);
+      expect(state[1][0]).toBe(32);
+      expect(state[1][1]).toBe(32);
+      expect(state[2][0]).toBe(64);
+      expect(state[2][1]).toBe(64);
+      expect(state[2][2]).toBe(128);
+      expect(state[3][0]).toBe(4);
+      expect(state[3][1]).toBe(4);
+      expect(state[3][2]).toBe(8);
```

Comment:
The test case 'should not merge multiple cells with the same value to the left multiple times during the move' is good, but it could also verify that the score is updated correctly after the merge, as this is an important part of the game's logic.